### PR TITLE
Differentiate between failures to load config file not existing vs. parse errors

### LINKED ISSUE #241: Invalid YAML removes most of the content of HLD
**Describe the bug:**
Adding an environment variable with no space between the key and the value removes most of the content of the HLD YAML file.

**To Reproduce:**
1. Make a /config/common.yaml  like this:
```
subcomponents:
  anotherservice:
    config:
      image:
        tag: "744"
      replicaCount: 1
  screenprofiler:
    config:
      envVar:
        important: true
        fabshouldfail: true
        is_this_a_bug: definitely
        culprit:"this is where it all goes wrong due to the lack of space between key and value"
      image:
        repository: somewheregood.azurecr.io/screenprofiler
        tag: "73"
      replicaCount: 1
```
The subcomponents | screenprofiler | envVar | culprit is the problem

2. Execute fabricate
```
./fab set --subcomponent screenprofiler image.tag=747
```
3. Fabricate removes most of the HLD file:
```
subcomponents:
  screenprofiler:
    config:
      image:
        tag: "747"
```

**Expected behavior:**
Fabricate should fail, as the YAML is invalid

OR

subcomponents | screenprofiler | image | tag is set to 747

Like 

```
subcomponents:
  anotherservice:
    config:
      image:
        tag: "744"
      replicaCount: 1
  screenprofiler:
    config:
      envVar:
        important: true
        fabshouldfail: true
        is_this_a_bug: definitely
        culprit:"this is where it all goes wrong due to the lack of space between key and value"
      image:
        repository: somewheregood.azurecr.io/screenprofiler
        tag: "747"
      replicaCount: 1
```

**Desktop (please complete the following information):**

Reproduced on:
 - WSL Windows 10 OS Build 18362.295
 - Azure DevOps Agent Hosted Ubuntu 1604

fab version 0.15.2

Execute via Azure DevOps Agent like so:
curl https://raw.githubusercontent.com/Microsoft/bedrock/master/gitops/azure-devops/build.sh > build.sh
chmod +x ./build.sh

curl https://raw.githubusercontent.com/Microsoft/bedrock/master/gitops/azure-devops/release.sh > release.sh
chmod +x ./release.sh

### PR DESCRIPTION
**Pull request by timfpark**
Fixes #241

### CODE REVIEW

File: core/componentConfig.go [hunk 1]
```diff
@@ -66,16 +66,20 @@ func (cc *ComponentConfig) MergeConfigFile(path string, environment string) (err
 func (cc *ComponentConfig) Load(environment string) (err error) {
 	err = cc.UnmarshalYAMLConfig(environment)
 
-	// fall back to looking for JSON if loading YAML fails.
-	if err != nil {
-		err = cc.UnmarshalJSONConfig(environment)
+	// If success or loading or parsing the file failed for reasons other than it didn't exist, return.
+	if err == nil || !os.IsNotExist(err) {
```
Comment (evanlouie) [inline core/componentConfig.go]: can you change to

```go
if err = err = cc.UnmarshalYAMLConfig(environment); err == nil || !os.IsNotExist(err) {
    return err
}
```

like to keep our style for immediate error returns consistent

File: core/componentConfig.go [hunk 2]
```diff
@@ -66,16 +66,20 @@ func (cc *ComponentConfig) MergeConfigFile(path string, environment string) (err
...
+	}
 
-		if err != nil {
-			// couldn't find any config files, so default back to yaml serialization
-			cc.Serialization = "yaml"
-		}
+	err = cc.UnmarshalJSONConfig(environment)
+
+	// If success or loading or parsing the file failed for reasons other than it didn't exist, return.
+	if err == nil || !os.IsNotExist(err) {
```
Comment (evanlouie) [inline core/componentConfig.go]: same as above

File: core/componentConfig.go (hunk 1 above)
Comment (timfpark) [reply core/componentConfig.go]: oh great catch - yes - will do

**Comment by timfpark** [comment]
@evanlouie Fixed your style changes - but also added checks during component loading since we don't want to hide parse failures there either.

**Pull request merged**